### PR TITLE
angularjs: changed $timeout signature, added tests

### PR DESCRIPTION
**Pull request by chrootsu**
https://docs.angularjs.org/api/ng/service/$timeout
https://github.com/angular/angular.js/blob/v1.4.3/src/ng/timeout.js

### CODE REVIEW

File: angularjs/angular-tests.ts [hunk 1]
```diff
@@ -287,6 +287,35 @@ var scope: ng.IScope = element.scope();
...
+        a: number;
+        b: string;
+        c: boolean;
+    }
+    var fnTResult: (...args: any[]) => TResult;
+    var promiseAny: angular.IPromise<any>;
+    var $timeout: angular.ITimeoutService;
+
+    // $timeout
+    result = <angular.IPromise<any>>$timeout();
```
Comment (vvakame) [inline angularjs/angular-tests.ts]: Why need type assertions?
I think type inference is better than type assertions.

Comment (chrootsu) [inline angularjs/angular-tests.ts]: @vvakame I believed that `result = $timeout()` only checks the possibility to call $timeout as function, but does not verify that the return type as expected, isn't it?

**Comment by chrootsu** [comment]
@vvakame Fixed.

**Comment by vvakame** [comment]
@chrootsu thanks mate!

**Pull request merged**